app/src/main/java/com/karumi/ui/view/adapter/SuperHeroViewHolder.kt

```diff
@@ -0,0 +1,44 @@
+package com.karumi.ui.view.adapter
+
+import android.content.Context
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import com.karumi.R
+import com.karumi.ui.domain.model.SuperHero
+import com.karumi.ui.presenter.SuperHeroesPresenter
+import com.squareup.picasso.Picasso
+
+class SuperHeroViewHolder(itemView: View,
+    private val presenter: SuperHeroesPresenter) : RecyclerView.ViewHolder(itemView) {
+    private val photoImageView: ImageView = itemView.findViewById(R.id.iv_super_hero_photo)
+    private val nameTextView: TextView = itemView.findViewById(R.id.tv_super_hero_name)
+    private val avengersBadgeView: View = itemView.findViewById(R.id.iv_avengers_badge)
+
+    private val context: Context
+        get() = itemView.context
+
+    fun render(superHero: SuperHero) {
+        hookListeners(superHero)
+        renderSuperHeroPhoto(superHero.photo)
+        renderSuperHeroName(superHero.name)
+        renderAvengersBadge(superHero.isAvenger)
+    }
+
+    private fun hookListeners(superHero: SuperHero) {
+        itemView.setOnClickListener { presenter.onSuperHeroClicked(superHero) }
+    }
+
+    private fun renderSuperHeroPhoto(photo: String) {
+        Picasso.with(context).load(photo).fit().centerCrop().into(photoImageView)
```

Comment:
Remember to check if the photo is defined. The Marvel API declares this attribute as optional but we are not handling it.

Reply:
I repeat the same question, if we manage the nullability here or in the api. And remember that the kata don't use internet the data comes from hardcoded info in the repositories.

Reply:
Choose the one you prefer. By definition the picture is optional. If I were you, I'd declare the photo as an optional and I'd check the definition of the photo value in this method, in the UI layer.

Reply:
I'm trying to know if it's a best option resolve the nullabilities as soon as we found, in this case the api layer or manage in the ui layer. :) I trying to learn because you have more experience with this kind of things.